spring-security/src/main/java/com/sap/cloud/security/spring/autoconfig/XsuaaTokenFlowAutoConfiguration.java

```diff
@@ -47,14 +56,44 @@
 	}
 
 	@Bean
-	@Conditional(PropertyConditions.class)
+	@Conditional({OnNotX509CredentialTypeCondition.class, PropertyConditions.class,})
 	public XsuaaTokenFlows xsuaaTokenFlows() {
 		logger.debug("auto-configures XsuaaTokenFlows.");
-		OAuth2ServiceEndpointsProvider endpointsProvider = new XsuaaDefaultEndpoints(xsuaaConfig.getUrl());
-		ClientCredentials clientCredentials = new ClientCredentials(xsuaaConfig.getClientId(),
-				xsuaaConfig.getClientSecret());
+		OAuth2ServiceEndpointsProvider endpointsProvider = new XsuaaDefaultEndpoints(xsuaaConfig);
+		ClientIdentity clientIdentity = xsuaaConfig.getClientIdentity();
 		OAuth2TokenService oAuth2TokenService = new DefaultOAuth2TokenService();
-		return new XsuaaTokenFlows(oAuth2TokenService, endpointsProvider, clientCredentials);
+		return new XsuaaTokenFlows(oAuth2TokenService, endpointsProvider, clientIdentity);
+	}
+
+	/**
+	 * Creates a new {@link XsuaaTokenFlows} bean that supports mTLS that
+	 * applications can auto-wire into their controllers to perform a programmatic
+	 * token flow exchange.
+	 *
+	 * - the {@link RestOperations} to use for the token flow exchange.
+	 *
+	 * @return the {@link XsuaaTokenFlows} API.
+	 */
+	@Bean
+	@ConditionalOnProperty(prefix = "sap.security.services.xsuaa", name = "credential-type", havingValue = "x509")
+	@ConditionalOnMissingBean
+	public XsuaaTokenFlows xsuaaMtlsTokenFlows() throws ServiceClientException {
+		logger.debug("auto-configures XsuaaTokenFlows using mTLS restOperations with uaacert endpoint: {}",
```

Comment:
I wonder whether this xsuaaMtlsTokenFlows would not also fit to non-mtls case? 
This depends on whether this requires applications to provide apache httpclient, while it worked previously without of apache rest client, when doing token-exchange.

Reply:
we need to separate it because of SpringHttpClient wold then require apache rest client, which shouldn't be the case for non-mtls case